funnel/utils/markdown/mdit_plugins/abbr.py

```diff
@@ -0,0 +1,173 @@
+"""
+Markdown-it-py plugin to introduce <abbr> markup for defined abbreviations.
+
+Ported from javascript plugin markdown-it-abbr.
+"""
+
+import re
+import unicodedata
+
+from markdown_it import MarkdownIt
+from markdown_it.rules_block import StateBlock
+from markdown_it.rules_inline import StateInline
+from markdown_it.token import Token
+
+__all__ = ['abbr_plugin']
+
+ASTERISK_CHAR = 0x2A  # ASCII value for `*`
+OPEN_SQBR_CHAR = 0x5B  # ASCII value for `[`
+CLOSE_SQBR_CHAR = 0x5D  # ASCII value for `]`
+BACKSLASH_CHAR = 0x5C  # ASCII value for `\`
+COLON_CHAR = 0x3A  # ASCII value for `:`
+
+PADDING_CHARACTERS = (
+    '['
+    + ''.join(
+        chr(x) for x in range(65536) if unicodedata.category(chr(x))[0] in ('P', 'Z')
+    )
+    .replace('\\', '\\\\')
+    .replace('[', '\\[')
+    .replace(']', '\\]')
+    + '\r\n$+<=>^`|~'
+    + ']'
+)
```

Comment:
Since this is in regex format and not a simple string of characters, the name should be `PADDING_CHARS_REGEX` or `PADDING_CHARS_RE` (we use `re` lowercase prefix or suffix for a compiled regex).